coordinate.cpp

```diff
@@ -49,35 +61,41 @@ void test_coordinate() {
 
     double default_speed = 1;
     c.move(direction);
-    assert(get_x(c) == default_speed);
-    assert(get_y(c) == 0 );
+    double error_x = abs(get_x(c) - default_speed);
+    double error_y = abs(get_y(c) - 0);
+    assert(error_x < 0.00001);
+    assert(error_y < 0.00001);
 
-    std::vector<double> speeds{0.1,0.5,2,-0.1,-0.2,-1};
+    std::vector<double> speeds{0.1, 0.5, 2, -0.1, -0.2, -1};
     for(const auto& speed : speeds)
       {
         c.reset_coords_to_zero();
         c.move(direction, speed);
-        assert(get_x(c) == speed );
-        assert(get_y(c) == 0 );
+        error_x = abs(get_x(c) - speed);
+        error_y = abs(get_y(c) - 0);
+        assert(error_x < 0.00001);
+        assert(error_y < 0.00001);
       }
 
     ///Test along y axis
     ///When direction is 0 player should
     ///only move along the y axis
-    direction = M_PI/2;
-
-    double default_speed = 1;
+    c.reset_coords_to_zero();
+    direction = M_PI / 2;
```

Comment:
We should use a vector of directions, like line 69, to test all the cases.

Reply:
I have added two more less trivial directions, but I think any more would be overkill, actually.
+ you need to make a separate expectation for every direction, unlike the convenient exemples of the x=0 or y = 0 movement 😅